python/paddle/v2/reader/decorator.py

```diff
@@ -224,3 +226,74 @@ def firstn_reader():
             yield item
 
     return firstn_reader
+
+
+class XmapEndSignal():
+    pass
+
+
+def xmap(mapper, reader, process_num, buffer_size):
+    """
+    Use multiprocess to map samples from reader by a mapper defined by user.
```

Comment:
Yes, it's here: https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/reader/decorator.py#L162

the `buffered` decorator will use a background thread to read the data. If you want map with multi-thread to speed up read, you can put a map decorator on top of the `buffered` decorator.